src/scripts/components/VideoForm.js

```diff
@@ -117,36 +117,51 @@ class VideoForm extends Component<Props, Object> {
   }
 
   componentWillReceiveProps (nextProps: Props): void {
-    this.setState(nextProps.selectedVideo)
-    this.setState({
-      id: nextProps.selectedVideo.id,
-      title: nextProps.selectedVideo.title,
-      description: nextProps.selectedVideo.description
-    })
+    const { selectedVideo } = nextProps
```

Comment:
The changes in non-reducer files were mainly bugs that were uncovered during work on this PR. Mostly that some props could be `undefined/null` in components and that case was not accounted for in the components themselves.